lib/matplotlib/backend_tools.py

```diff
@@ -0,0 +1,927 @@
+        `matplotlib.backend_bases.NavigationBase.remove_tool`
+        """
+        pass
+
+
+class ToolToggleBase(ToolBase):
+    """Toggleable tool
+
+    Every time it is triggered, it switches between enable and disable
+    """
+
+    radio_group = None
+    """Attribute to group 'radio' like tools (mutually exclusive)
+
+    **String** that identifies the group or **None** if not belonging to a
+    group
+    """
+
+    cursor = None
+    """Cursor to use when the tool is active"""
+
+    def __init__(self, *args, **kwargs):
+        ToolBase.__init__(self, *args, **kwargs)
+        self._toggled = False
+
+    def trigger(self, sender, event, data=None):
+        """Calls `enable` or `disable` based on `toggled` value"""
+        if self._toggled:
+            self.disable(event)
+        else:
+            self.enable(event)
+        self._toggled = not self._toggled
+
+    def enable(self, event=None):
+        """Enable the toggle tool
+
+        `trigger` calls this method when `toggled` is False
+        """
+
+        pass
+
+    def disable(self, event=None):
+        """Disable the toggle tool
+
+        `trigger` call this methond when `toggled` is True.
+
+        This can happen in different circumstances
+
+        * Click on the toolbar tool button
+        * Call to `matplotlib.backend_bases.NavigationBase.tool_trigger_event`
+        * Another `ToolToggleBase` derived tool is triggered
+          (from the same `Navigation`)
+        """
+
+        pass
+
+    @property
+    def toggled(self):
+        """State of the toggled tool"""
+
+        return self._toggled
+
+
+class SetCursorBase(ToolBase):
+    """Change to the current cursor while inaxes
+
+    This tool, keeps track of all `ToolToggleBase` derived tools, and calls
+    set_cursor when a tool gets triggered
+    """
+    def __init__(self, *args, **kwargs):
+        ToolBase.__init__(self, *args, **kwargs)
+        self._idDrag = self.figure.canvas.mpl_connect(
+            'motion_notify_event', self._set_cursor_cbk)
+        self._cursor = None
+        self._default_cursor = cursors.POINTER
+        self._last_cursor = self._default_cursor
+        self.navigation.nav_connect('tool_added_event', self._add_tool_cbk)
+
+        # process current tools
+        for tool in self.navigation.tools.values():
+            self._add_tool(tool)
+
+    def _tool_trigger_cbk(self, event):
+        if event.tool.toggled:
+            self._cursor = event.tool.cursor
+        else:
+            self._cursor = None
+
+        self._set_cursor_cbk(event.canvasevent)
+
+    # If the tool is toggleable, set the cursor when the tool is triggered
+    def _add_tool(self, tool):
+        if getattr(tool, 'cursor', None) is not None:
+            self.navigation.nav_connect('tool_trigger_%s' % tool.name,
+                                        self._tool_trigger_cbk)
+
+    # If tool is added, process it
+    def _add_tool_cbk(self, event):
+        if event.tool is self:
```

Comment:
the `tool-added-event` is fired after the tool is instantiated, to it will get called once